src/hevm/src/EVM/UnitTest.hs

```diff
@@ -546,24 +547,22 @@ symRun opts@UnitTestOptions{..} concreteVm testName types = do
           case view result vm' of
             Just (VMSuccess (SymbolicBuffer buf)) ->
               constrain $ litBytes (encodeAbiValue $ AbiBool $ not shouldFail) .== buf
-            _ -> error "unexpected return value"
+            r -> error $ "unexpected return value: " ++ show r
         checkSat >>= \case
           Sat -> do
-            prettyCd <- prettyCalldata cd testName types
-            let explorationFailed = case (view result vm') of
+            prettyCd <- prettyCalldata cd' testName types
+            let explorationFailed = case view result vm' of
                   Just (VMFailure e) -> case e of
-                                          NotUnique -> True
+                                          NotUnique _ -> True
                                           UnexpectedSymbolicArg -> True
-                                          DeadPath -> True
                                           _ -> False
                   _ -> False
             return $
               if shouldFail && bailed && not explorationFailed
               then Right ()
               else Left (vm', prettyCd)
           Unsat -> return $ Right ()
-          Unk -> let out = vm' & set result (Just (VMFailure SMTTimeout))
-                 in  return $ Left (out, "unknown")
+          Unk -> return $ Left (vm', "unknown; query timeout")
```

Comment:
why? now this shows:

```
  Counterexample:

    result:   Failed: DSTest Assertion Violation
    calldata: unknown; query timeout
```

Instead of:

```
  Counterexample:

    result:   Failed: SMTTimeout
    calldata: unknown
```

Which I find a little less clear.

Reply:
or it shows:
```
 Counterexample:

    result:   Failed: revert("ds-math: mul overflow")
    calldata: unknown; query timeout

```

Reply:
ah ok, hadn't considered that. I'm fine with this then :+1: